pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -85,14 +87,72 @@ func validateKubeSchedulerProfile(path *field.Path, profile *config.KubeSchedule
 	if len(profile.SchedulerName) == 0 {
 		allErrs = append(allErrs, field.Required(path.Child("schedulerName"), ""))
 	}
+	allErrs = append(allErrs, validatePluginConfig(profile)...)
+	return allErrs
+}
+
+func fromAggregate(agg utilerrors.Aggregate) field.ErrorList {
+	errs := agg.Errors()
+	list := make(field.ErrorList, len(errs))
+	for i := range errs {
+		list[i] = errs[i].(*field.Error)
+	}
+	return list
+}
+
+func validatePluginConfig(profile *config.KubeSchedulerProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+	m := map[string]interface{}{
+		"DefaultPreemption":           ValidateDefaultPreemptionArgs,
+		"InterPodAffinity":            ValidateInterPodAffinityArgs,
+		"NodeAffinity":                ValidateNodeAffinityArgs,
+		"NodeLabel":                   ValidateNodeLabelArgs,
+		"NodeResourcesLeastAllocated": ValidateNodeResourcesLeastAllocatedArgs,
+		"NodeResourcesMostAllocated":  ValidateNodeResourcesMostAllocatedArgs,
+		"PodTopologySpread":           ValidatePodTopologySpreadArgs,
+		"RequestedToCapacityRatio":    ValidateRequestedToCapacityRatioArgs,
+		"VolumeBinding":               ValidateVolumeBindingArgs,
+		//"NodeResourcesFitArgs":        ValidateNodeResourcesFitArgs, // TODO: unmask this line when #100477 got merged.
+	}
+
+	pluginConfig := make(map[string]bool, len(profile.PluginConfig))
+	for i := range profile.PluginConfig {
+		name := profile.PluginConfig[i].Name
+		args := profile.PluginConfig[i].Args
+		if pluginConfig[name] {
+			allErrs = append(allErrs, field.Duplicate(field.NewPath("pluginConfig"), name))
+		} else {
+			pluginConfig[name] = true
+		}
+		if validateFunc, ok := m[name]; ok {
+			in := []reflect.Value{reflect.ValueOf(args)}
+			res := reflect.ValueOf(validateFunc).Call(in)
+			if errList, ok := res[0].Interface().(utilerrors.Aggregate); ok {
```

Comment:
The problem is that I need to build a `field.ErrorList`, while the validation function return either `Aggregate` or just a `field.Error` which means we cannot append it directly.

`errors.Flatten`  give us a `Aggregate`  instead of `field.ErrorList`, this is why I need to parse the the `Aggregate` to a error list.

Reply:
you can return an `Aggregate` instead.

You can append `Aggregate` or `field.Error`, no problem.  `Flatten` would do the flattening for you, after the fact.
In fact, you can choose to do it only once at the end in `validateKubeSchedulerProfile`

Reply:
@alculquicondor , updated so that the function will be called only once when all profiles are validated, pls kindly check it again.

Reply:
You don't have to check for the type, just append

Reply:
It is not just type check,  as appending  `reflect.Value` or `interface{}` to a `error` list is not allowed, we need the assertion to convert it to an `error`. I will simplify the code here in the next iteration.

Reply:
sure, then just cast to an error